data/common.yaml

```diff
@@ -14,6 +14,102 @@ gpfs::add_client::ssh_private_key_path: "/root/.ssh/id_gpfs"
 gpfs::add_client::ssh_public_key_type: "rsa"
 gpfs::add_client::script_tgt_fn: "/root/gpfs_add_client.sh"
 gpfs::cron::accept_license: false
+gpfs::health::enabled: false
+gpfs::health::file_base_name: "gpfs_client_health"
+gpfs::health::telegraf_cfg:
+  data_format: "influx"
+  interval: "1m"
+  timeout: "30s"
+gpfs::health::telegraf_script: |2+
```

Comment:
I feel like keeping scripts in separate files fits better with how my brain thinks.  I can really only speak on the Telegraf side (and not all the other things you all are doing with puppet) but I try really hard to avoid (and I think I've done so successfully) there being anything dynamic (per cluster/deployment scenario) in these custom exec scripts.  Anything custom is source'd from a config file I believe is already in hiera which that part makes sense to me, and if anything needed to be dynamic in the script itself I'd just tack another variable into the config file.  

When doing development I usually work outside of any repo, for example if adding a feature to a slurm check I go to a scheduler and make a copy of the current running script, and then edit that copy (since Telegraf won't be trying to execute that copied-aside version).  I then execute it by hand when I want to see how it does/is it working as expected as I'm writing the new feature (or patching a bug).  Once I have the new working/tested version of the check,  my current workflow is to then copy the script in it's entirety, open the script file in the puppet repo, nuke the entire contents of that file, then paste in the new version.  That way I know things match 100% between what I tested as working and what I'm asking folks to review in git/push to production.  With the script in hiera, I think I could technically do most of that, but I'd have to make sure I just nuke the script part of hiera and not anything else, which I agree most of the time isn't hard but I'd be more worried about accidentally deleting things I shouldn't.